0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+relation type. Subtype will inherit all owned attribute types and all roles related or played by the parent type.
+
+<!-- test-ignore -->
+```typeql
+define
+
+membership sub relation, relates parent, relates member;
+collection-membership sub membership;
+```
+
+In the example above, the `collection-membership` relation type inherits the `parent` and `member` roles defined in 
+its parent type, `membership`.
+
+The names of the inherited roles can be overridden to distinguish between the roles inherited by a relation subtype 
+vs. the roles defined by its parent type. However, the relation type being subtyped must be abstract. 
+It is not possible to override roles inherited by concrete relation types.
+
+<!-- test-ignore -->
+```typeql
+define
+
+membership sub relation, abstract, relates parent, relates member;
+
+collection-membership sub membership, 
+relates parent-collection as parent, 
+relates collection-member as member;
+```
+
+In the example above, the `collection-membership` relation type subtypes the `membership` relation type, and overrides 
+the inherited `parent` role as `parent-collection` and the inherited `member` role as `collection-member`.
+
+<div class="note">
+[Note]
+The two examples above can be run back to back. The second one will update the `membership` relation type to make it 
+abstract, and the `collection-membership` type to override its inherited roles.
+</div>
+
+##### Complex example
+
+<!-- test-ignore -->
+```typeql
+define
+
+ownership-type sub attribute, value string;
+
+ownership sub relation, abstract, 
+relates owned, 
+relates owner;
+
+group-ownership sub ownership, 
+relates owned-group as owned, 
+relates group-owner as owner, 
+owns ownership-type;
+
+object-ownership sub ownership, 
+relates owned-object as owned, 
+relates object-owner as owner, 
+owns ownership-type;
+
+access sub relation, 
+relates accessed-object, 
+relates valid-action, 
+plays change-request:requested-change;
+
+change-request sub relation, 
+relates requesting-subject, 
+relates requested-subject, 
+relates requested-change;
+```
+
+The example above defines one attribute type and five relation types:
+
+* `ownership` — subtypes the base `relation` type, is abstract, and relates `owned` and `owner` roles.
+* `group-ownership` — subtypes `ownership` relation type, relates `owned-group` as `owned` and `group-owner` as `owner`.
+* `object-ownership` — subtypes `ownership` relation, relates `owned-object` as `owned` and `object-owner` as `owner`.
+* `access` — subtypes the `relation` base type, relates `accessed-object` (e.g., files) and `valid-actions` (e.g., 
+  read), plays the role of `requested-change` in `change-request` relation type.
+* `change-request` — subtypes the `relation` base type, relates `requesting-subject`, `requested-subject` and
+  `requested-change`.
+
+### Define rules
+
+#### Overview
+
+Rules are defined independently of any types.
```

Comment:
Unsure what you mean - Any type referenced in the rule has to be defined.
`rule my-rule: when { $p isa non-existent-type; ...} then { ... };` will throw an error

Reply:
That is what we declare for every base type (as they are independent). And the same is true for rules. Nobody is saying that you don't need to define types used in rules. If you want I can add a Note about that.